json2csv.py

```diff
@@ -0,0 +1,65 @@
+"""Convert JSON to CSV for open_parliament_at source."""
+
+import click
+import json
+import csv
+import logging
+import re
+
+
+logger = logging.getLogger(__name__)
+
+WANTED_FIELDS = ['first_name', 'last_name', 'salutation', 'title', 'political_affiliation',
+                 'state', 'occupation', 'place_of_birth', 'date_of_birth']
+
+party_extract_re = re.compile(r'^.* \((?P<party>\w+)\)$')
+
+@click.command()
+@click.argument('input', type=click.File('rb'), default='-')
+@click.argument('output', type=click.File('w'), default='-')
+def convert_to_csv(input, output):
+    scraped = json.load(input)
+
+    def sort_by_id(item):
+        return int(item["id"])
+    sorted_input = sorted(scraped, key=sort_by_id)
+
+    contacts = []
+    for mp_data in sorted_input:
+        mp = {
+            'identifier': mp_data['id']
+        }
+        emails = mp_data.pop('emails')
+        if len(emails) > 0:
+            mp['email'] = emails[0]
+        else:
+            mp['email'] = ''
+            logger.warning('Contact without email: %s (%s %s)', mp_data['id'], mp_data['first_name'], mp_data['last_name'])
+        mp.update({k: mp_data[k] for k in WANTED_FIELDS})
+        committees = sorted([c['name'] for f in mp_data['committees'].values() for c in f.values()])
+        mp['committees'] = ',' + ','.join(committees) + ','
+        mandates = [m['title'] for m in mp_data['mandates']]
+        # extract party from political_affiliation field
+        party = party_extract_re.match(mp_data['political_affiliation'])
+        if party and party.group('party'):
+            mp['party'] = party.group('party')
+        else:
+            mp['party'] = ''
+            logger.warning('Contact without party: %s (%s %s)', mp_data['id'], mp_data['first_name'], mp_data['last_name'])
+        mp['mandates'] = ',' + ','.join(mandates) + ','
+        contacts.append(mp)
+
+    def get_values(obj, keys):
+        return [obj[k] for k in keys]
+
+    csvwriter = csv.writer(output)
+    output_mapping = ['identifier', 'first_name', 'last_name', 'email', 'salutation', 'title', 'party',
+                      'political_affiliation', 'state', 'occupation', 'place_of_birth', 'date_of_birth',
+                      'mandates', 'committees']
+    csvwriter.writerow(output_mapping)
+    for item in contacts:
+        csvwriter.writerow(get_values(item, output_mapping))
```

Comment:
You could use a `csv.DictWriter(extrasaction='ignore')` in this case then you wouldn’t need `get_value()`.